packages/pigeon/mock_handler_tester/test/message.dart

```diff
@@ -1,173 +1,148 @@
           'dev.flutter.pigeon.FlutterSearchApi.search', StandardMessageCodec());
-      channel.setMessageHandler((dynamic message) async {
-        final Map<dynamic, dynamic> mapMessage =
-            message as Map<dynamic, dynamic>;
-        final SearchRequest input = SearchRequest._fromMap(mapMessage);
-        final SearchReply output = api.search(input);
-        return output._toMap();
-      });
+      if (api == null) {
+        channel.setMessageHandler(null);
+      } else {
+        channel.setMessageHandler((Object message) async {
+          if (message == null) {
+            return null;
+          }
+          final SearchRequest input = SearchRequest.decode(message);
+          final SearchReply output = api.search(input);
+          return output.encode();
+        });
+      }
     }
   }
 }
 
 class NestedApi {
   Future<SearchReply> search(Nested arg) async {
-    final Map<dynamic, dynamic> requestMap = arg._toMap();
-    const BasicMessageChannel<dynamic> channel = BasicMessageChannel<dynamic>(
+    final Object encoded = arg.encode();
+    const BasicMessageChannel<Object> channel = BasicMessageChannel<Object>(
         'dev.flutter.pigeon.NestedApi.search', StandardMessageCodec());
-
-    final Map<dynamic, dynamic> replyMap = await channel.send(requestMap);
+    final Map<Object, Object> replyMap =
+        await channel.send(encoded) as Map<Object, Object>;
     if (replyMap == null) {
       throw PlatformException(
-          code: 'channel-error',
-          message: 'Unable to establish connection on channel.',
-          details: null);
+        code: 'channel-error',
+        message: 'Unable to establish connection on channel.',
+        details: null,
+      );
     } else if (replyMap['error'] != null) {
-      final Map<dynamic, dynamic> error = replyMap['error'];
+      final Map<Object, Object> error =
+          replyMap['error'] as Map<Object, Object>;
       throw PlatformException(
-          code: error['code'],
-          message: error['message'],
-          details: error['details']);
+        code: error['code'] as String,
+        message: error['message'] as String,
+        details: error['details'],
+      );
     } else {
-      return SearchReply._fromMap(replyMap['result']);
-    }
-  }
-}
-
-abstract class TestNestedApi {
-  SearchReply search(Nested arg);
-  static void setup(TestNestedApi api) {
-    {
-      const BasicMessageChannel<dynamic> channel = BasicMessageChannel<dynamic>(
-          'dev.flutter.pigeon.NestedApi.search', StandardMessageCodec());
-      channel.setMockMessageHandler((dynamic message) async {
-        final Map<dynamic, dynamic> mapMessage =
-            message as Map<dynamic, dynamic>;
-        final Nested input = Nested._fromMap(mapMessage);
-        final SearchReply output = api.search(input);
-        return <dynamic, dynamic>{'result': output._toMap()};
-      });
+      return SearchReply.decode(replyMap['result']);
```

Comment:
Here's the call whose behavior would change as a result of removing the null check in decode.

Reply:
Ah, yes. In null-safe mode this has a `!` in the new generated code. I can check it for a `result` key and throw if it's missing, if you like. Or do something else. Your call.